### PR TITLE
Working on internationalization

### PR DESCRIPTION
**Pull request by danbilokha**

### CODE REVIEW

File: server/src/messages/feature/settingsMessages.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+): string =>
+    getLocalizedMessage(locale, [
+        'Language is not supported 😔',
+        '. Choose from the list',
+    ]).join();
+
+export const cannotSetupLanguageMessage = (locale: string): string =>
+    getLocalizedMessage(locale, [
+        `Sorry, I cannot setup the ${locale} for you. I am really sorry  🙇🏽`,
+        `. Probably my database is out of reach. Try latter`,
```
Comment (Proladge) [inline server/src/messages/feature/settingsMessages.ts:17]: `Sorry, I cannot setup the ${locale} for you. I am really sorry  🙇🏽`,
        `. Probably my database is out of reach. Please try a bit latter`

@danbilokha please change message to be

File: server/src/services/domain/localization.service.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import * as i18n from 'i18n';
+
+export const getLocalizedMessage = (
```
Comment (Proladge) [inline server/src/services/domain/localization.service.ts:3]: It's actually difficult to understand. What does it mean here when we say we want to get localized message not only by name but by other parameters,  @danbilokha  could you please bring up an example

File: server/src/services/domain/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        return this.storage.addUser({
+            ...user,
+            settings: {
+                ...user.settings,
+                locale,
+            },
+        });
+    }
+
+    public async setUserInterruptedCommand(
```
Comment (Proladge) [inline server/src/services/domain/user.service.ts:42]: @danbilokha what is interrupted command?

File: server/src/bots/telegram/services/registry/withSingleParameterAfterCommand.ts [hunk 4]
```diff
@@ -12,37 +11,39 @@ import { LogCategory } from '../../../../models/constants';
...
             logger.error(
                 `Error happend inside withSingleParameterAfterCommand() for ${chatId} with message: ${message.text} and ikCbData: ${ikCbData}`,
                 err,
                 LogCategory.Command,
-                chatId
+                chatId,
             );
 
-            return noResponse(bot, message, chatId);
+            return noResponse({ bot, message, chatId, user, messageHandlerRegistry });
```
Comment (Proladge) [inline server/src/bots/telegram/services/registry/withSingleParameterAfterCommand.ts:46]: why do we send whole registry to response layer?

File: server/src/bots/telegram/index.ts [hunk 5]
```diff
@@ -65,62 +62,88 @@ export function runTelegramBot(
...
                 UserMessages.Unsubscribe,
             ],
-            unsubscribeStrategyResponse
+            withSingleParameterAfterCommand(
+                unsubscribeStrategyResponse,
+            ),
         )
         .registerMessageHandler(
             [UserRegExps.Trends],
-            withTwoArgumentsAfterCommand(trendsByCountryResponse)
```
Comment (Proladge) [inline server/src/bots/telegram/index.ts:116]: I was thinking If we could wrap ```withSingleParameterAfterCommand``` with ```withTwoArgumentsAfterCommand```
That way we won't need to write here both of the decorators. just ```withTwoArgumentsAfterCommand```
@danbilokha what do you think?

File: server/src/bots/telegram/botResponse/runInterruptedCommandReponse.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import * as TelegramBot from 'node-telegram-bot-api';
```
Comment (Proladge) [inline server/src/bots/telegram/botResponse/runInterruptedCommandReponse.ts:1]: @danbilokha what is interrupted command?

File: server/src/locales/en.json [hunk 7]
```diff
@@ -0,0 +1,25 @@
+{
```
[the remark this answers is not in the document]
Comment (danbilokha) [reply server/src/locales/en.json:1]: maybe you're right. however, I didn't invest time in that and I think that it's just easier to translate when you have this type of keys

File: server/src/bots/telegram/services/registry/messageHandlerRegistry.ts [hunk 8]
```diff
@@ -42,109 +43,144 @@ export class MessageHandlerRegistry {
...
+    }
+
+    private async createAndAddUser(
+        message: TelegramBot.Message,
+        chatId: number,
+    ): Promise<User> {
+        return telegramUserService.addUser({
+            ...DEFAULT_USER_SETTINGS,
+            chatId,
+            userName: message.chat.username || '',
```
[the remark this answers is not in the document]
Comment (danbilokha) [reply server/src/bots/telegram/services/registry/messageHandlerRegistry.ts:180]: discussed

File: server/src/bots/telegram/services/registry/withTwoArgumentsAfterCommand.ts [hunk 9]
```diff
@@ -1,45 +1,48 @@
...
             logger.error(
                 `Error happend inside withTwoArgumentsAfterCommand() for ${chatId} with message: ${message.text} and ikCbData: ${ikCbData}`,
                 err,
                 LogCategory.Command,
-                chatId
+                chatId,
             );
 
-            return noResponse(this.bot, message, chatId);
+            return noResponse({ bot, message, user, chatId, messageHandlerRegistry });
```
[the remark this answers is not in the document]
Comment (danbilokha) [reply server/src/bots/telegram/services/registry/withTwoArgumentsAfterCommand.ts:37]: will take a look

File: server/src/messages/feature/settingsMessages.ts (hunk 1 above)
Comment (danbilokha) [reply server/src/messages/feature/settingsMessages.ts:17]: which one? :D

File: server/src/bots/telegram/services/registry/withSingleParameterAfterCommand.ts (hunk 4 above)
Comment (danbilokha) [reply server/src/bots/telegram/services/registry/withSingleParameterAfterCommand.ts:46]: will try to remove

File: server/src/bots/telegram/index.ts (hunk 5 above)
Comment (danbilokha) [reply server/src/bots/telegram/index.ts:116]: discussed

File: server/src/bots/telegram/botResponse/runInterruptedCommandReponse.ts (hunk 6 above)
Comment (danbilokha) [reply server/src/bots/telegram/botResponse/runInterruptedCommandReponse.ts:1]: discussed

File: server/src/services/domain/localization.service.ts (hunk 2 above)
Comment (danbilokha) [reply server/src/services/domain/localization.service.ts:3]: added example

File: server/src/services/domain/user.service.ts (hunk 3 above)
Comment (danbilokha) [reply server/src/services/domain/user.service.ts:42]: discussed

File: server/src/messages/feature/settingsMessages.ts (hunk 1 above)
Comment (Proladge) [reply server/src/messages/feature/settingsMessages.ts:17]: The one I've posted. It jsut looks more polite when adding please to it

File: server/src/bots/telegram/services/registry/messageHandlerRegistry.ts [hunk 10]
```diff
@@ -164,22 +198,36 @@ export class MessageHandlerRegistry {
...
-        chatId: number,
+        chatId: number
     ): Promise<User> {
         return telegramUserService.addUser({
             ...DEFAULT_USER_SETTINGS,
             chatId,
-            userName: message.chat.username || '',
-            firstName: message.chat.first_name || '',
-            lastName: message.chat.last_name || '',
+            userName: message.chat?.username,
```
[the remark this answers is not in the document]
Comment (danbilokha) [reply server/src/bots/telegram/services/registry/messageHandlerRegistry.ts:228]: just reverted it back

**Pull request merged**